storage/redis_adapter.go

```diff
@@ -29,11 +29,11 @@ type messagePipeliner struct {
 	errCh         chan error
 }
 
-func newMessagePipeliner(bufferSize int, redisClient *r.Client, errCh chan error) *messagePipeliner {
+func newMessagePipeliner(bufferSize int, redisClient *r.Client, timeoutSeconds int, errCh chan error) *messagePipeliner {
```

Comment:
So... I did try having a `time.Duration` in the `redisConfig` type, but found that envconfig couldn't handle that. So, barring that, the only way to pass a `time.Duration` to this function would be for any caller to convert the `int` to a `time.Duration` before calling. Somehow I'm not sure if that's actually better, but I am certain I could be convinced.